src/models/transferSettlement/facade.js

```diff
@@ -333,8 +334,25 @@ async function getSettlementModelByTransferId (transferId, settlementGranularity
     .select('settlementModel.*')
   if (settlementModelByTransferId.length === 0) {
     const allSettlementModels = await Db.from('settlementModel').find()
-    const defaultGrossSettlementModel = allSettlementModels.filter(sm => (sm.currencyId === null && sm.settlementGranularityId === SettlementEnum.SettlementGranularity.GROSS))
-    return defaultGrossSettlementModel
+    const transferCurrency = (await TransferFacade.getByIdLight(transferId)).currencyId
+    switch (settlementGranularityName) {
+      case SettlementEnum.settlementGranularityName.GROSS: {
+        const netModelWithCurrency = allSettlementModels.filter(sm => (sm.currencyId === transferCurrency && sm.settlementGranularityId === SettlementEnum.SettlementGranularity.NET))
+        if (netModelWithCurrency.length === 0) {
+          const defaultGrossSettlementModel = allSettlementModels.filter(sm => (sm.currencyId === null && sm.settlementGranularityId === SettlementEnum.SettlementGranularity.GROSS))
+          return defaultGrossSettlementModel
+        }
+        break
+      }
+      case SettlementEnum.settlementGranularityName.NET: {
```

Comment:
Is there any reason that you need to check for GROSS within the Net case?

I think you only need to do this:

```suggestion
      case SettlementEnum.settlementGranularityName.NET: {
        return allSettlementModels.filter(sm => (sm.currencyId === transferCurrency && sm.settlementGranularityId === SettlementEnum.SettlementGranularity.NET))
      }
```

Reply:
yes, to not look for default, in case of the transfer is in currency that has the opposite model.